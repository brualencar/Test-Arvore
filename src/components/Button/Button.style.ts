import { styled, css } from 'styled-components';

import { StyledButtonProps } from './Button.types';

const COLOR = {
  primary: css`
    color: ${({ theme }) => theme.colors.text.neutral.light};
    background: ${({ theme }) => theme.colors.background.neutral.white};
    border: 1px solid ${({ theme }) => theme.colors.text.neutral.light};
  `,
  secondary: css`
    color: ${({ theme }) => theme.colors.text.neutral.white};
    background: ${({ theme }) => theme.colors.background.neutral.strong};
    border: 1px solid rgba(64, 106, 118, 0.2);
  `,
  tertiary: css``,
};

export const Button = styled.button<StyledButtonProps>`
  height: 40px;
  border-radius: 10px;
  padding: ${({ theme }) => theme.padding.km3}
    ${({ theme }) => theme.padding.km4};
  ${({ variantColor }) =>
    css`
      ${COLOR[variantColor]}
    `};

  ${({ isUppercase }) =>
    isUppercase &&
    css`
      text-transform: uppercase;
    `}
`;
