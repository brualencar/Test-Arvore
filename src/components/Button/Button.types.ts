import { ReactElement } from 'react';

type Variant = 'primary' | 'secondary' | 'tertiary';

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  color?: string;
  text?: string;
  children?: ReactElement;
  variant: Variant;
  isUppercase: boolean;
}

export interface StyledButtonProps {
  variantColor: Variant;
  isUppercase: boolean;
}
