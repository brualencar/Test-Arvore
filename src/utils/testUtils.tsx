import React, { PropsWithChildren } from 'react';

import { ThemeProvider } from 'styled-components';

import { render as rtlRender, RenderOptions } from '@testing-library/react';

import { theme } from '../assets/styles/globalStyle';

function render(ui: React.ReactElement, renderOptions?: RenderOptions) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
