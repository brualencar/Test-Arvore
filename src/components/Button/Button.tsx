import React from 'react';

import * as Styled from './Button.style';
import { ButtonProps } from './Button.types';

function Button({
  color,
  text,
  variant,
  isUppercase = false,
  onClick,
}: ButtonProps) {
  return (
    <Styled.Button
      type='button'
      onClick={onClick}
      color={color}
      $variantColor={variant}
      $isUppercase={isUppercase}>
      {text}
    </Styled.Button>
  );
}

export default Button;
