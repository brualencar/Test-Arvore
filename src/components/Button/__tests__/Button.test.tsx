import React from 'react';

import { render } from '../../../utils/testUtils';
import Button from '../Button';

describe('Buttoon', () => {
  it('should render snapshot', () => {
    const { container } = render(<Button variant='primary' />);
    expect(container).toMatchSnapshot();
  });
});
