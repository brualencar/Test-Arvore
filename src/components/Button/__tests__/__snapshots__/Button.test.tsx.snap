// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttoon should render snapshot 1`] = `
<div>
  <button
    class="sc-bdDrbm dGflXy"
    type="button"
  />
</div>
`;
